### PR TITLE
Add localization to error messages from ExceptionTranslationFilter

### PR DESCRIPTION
**Pull request by britter**
In by simple Spring Boot application with Basic Authentication, I can add localization and custom messages if somebody tries to access a resource with wrong credentials. This is implemented in `AbstractUserDetailsAuthenticationProvider`.

However when somebody forgets to add authentication information the `ExceptionTranslationFilter` will throw an `InsufficientAuthenticationException`. Since the message is hard coded, there is no easy way to customize or localize the error message.

This PR adds localization using message bundles and a `MessageSourceAccessor` the same way as it is implemented in `AbstractUserDetailsAuthenticationProvider`.

### CODE REVIEW

**Comment by britter** [comment]
@jgrandja any feedback?

File: web/src/main/java/org/springframework/security/web/access/ExceptionTranslationFilter.java [hunk 1]
```diff
@@ -83,6 +86,8 @@
 
 	private RequestCache requestCache = new HttpSessionRequestCache();
 
+	private MessageSourceAccessor messages = SpringSecurityMessageSource.getAccessor();
```
Comment (jgrandja) [inline web/src/main/java/org/springframework/security/web/access/ExceptionTranslationFilter.java]: Add final modifier

File: web/src/main/resources/org/springframework/security/web/messages_en.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+ExceptionTranslationFilter.insufficientAuthentication=Full authentication is required to access this resource
```
Comment (jgrandja) [inline web/src/main/resources/org/springframework/security/web/messages_en.properties]: Add this key to messages.properties (default bundle) as well

Comment (britter) [reply web/src/main/resources/org/springframework/security/web/messages_en.properties]: Done!

**Comment by britter** [comment]
I need some time to work on a test, since I have to understand how the current tests work and how to make the test reliable on different system locales. Thank you!

**Comment by britter** [comment]
@jgrandja all done. Let's see what Travis thinks.

**Comment by rwinch** [comment]
Thanks for the PR @britter! This is now merged into master via fffd781

**Pull request closed without merging**